lib/inferno/dsl/fhir_evaluation/rules/all_resources_reachable.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_relative '../../../utils/evaluator_util'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        class AllResourcesReachable < Rule
+          attr_accessor :config
+
+          def check(context)
+            @config = context.config
+            # TODO: add some customizable configurations for this rule:
+            # - whether it should be checked at all
+            # - determine the base/root resource
+            # - which types/resources to ignore
+            # TODO: for the customizable configurations, use the config settings below:
+            # config.Rule.AllResourcesReachable.CheckedAtAll: true/false
+            # config.Rule.AllResourcesReachable.RunOnlyBaseResource: true/false
+            # config.Rule.AllResourcesReachable.IgnoreType: - value
+            # TODO: can come up with a "connectedness metric" to see how well-connected the data set is
+
+            # every resource is either making a resolvable reference or is referenced
+            util = Inferno::Utils::EvaluatorUtil
+            @referenced_resources = Set.new
+            @referencing_resources = Set.new
+            @resourcetype_ids, @resource_type_ids, @resource_ids, references = util.extract_ids_references(context.data)
+            references.each do |id, refs|
+              assess_reachability(id, refs)
+            end
+
+            island_resources = @resource_ids - @referenced_resources - @referencing_resources
+            sorted_island_resources = []
```

Comment:
Are these actually sorted? I don't see any sorting anywhere.

Reply:
Why is it being converted back to a Set after sorting?

Reply:
Sorting was added later and I wanted to be minimum invasive to the original data types as I am not the one who wrote this algorithm at the beginning.

Reply:
So I simplified it by https://github.com/inferno-framework/inferno-core/pull/583/commits/808828b64c574dfae6a9d55573499a2cd02b44e3 Removed creating another Set to put it back to the original data type.

Not sure below is a good Ruby practice, since it's the same variable the data type changed from Set to Array in the end. Let me know if this isn't.

`island_resources.to_a.sort!`